deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json

```diff
@@ -0,0 +1,34 @@
+{
+  "Name": "Upgrade Chocolatey for Business Client",
+  "ScheduledDateTimeUtc": null,
+  "LastScheduledDateTimeUtc": null,
+  "RepeatPeriod": 0,
+  "ExecutionTimeoutInSeconds": 14400,
+  "RequiresApproval": false,
+  "DeploymentSteps": [
+    {
+      "Name": "Upgrade Chocolatey for Business Client Components",
+      "IsPrivileged": true,
+      "PlanOrder": 1,
+      "Script": "$delayInMinutes = 1\n\n# If using an internal repository to install Chocolatey Agent, replace `chocolatey.licensed` below\n# with the name or URL of your internally configured source.\nchoco upgrade chocolatey-agent --source 'chocolatey.licensed'\n\n# Ensure the deployment task registers as failed if the installation failed, and skip registering the\n# scheduled task.\nif ($LASTEXITCODE -ne 0) {\n    Write-Error 'The upgrade failed!'\n    exit $LASTEXITCODE\n}\n\n# Restart the Agent service after the preset delay time via a scheduled task.\n$ScheduledTask = @{\n    TaskName = \"restart chocolatey-agent\"\n    Description = \"Upgrade Chocolatey Agent\"\n    Trigger = New-ScheduledTaskTrigger -Once -At (Get-Date).AddMinutes($delayInMinutes)\n    Action = New-ScheduledTaskAction -Execute 'powershell.exe' -Argument \"-WindowStyle Hidden -Command Restart-Service chocolatey-agent\"\n    Principal = New-ScheduledTaskPrincipal -GroupId Administrators -RunLevel Highest\n    Settings = New-ScheduledTaskSettingsSet -Hidden\n}\nRegister-ScheduledTask @ScheduledTask -Verbose:$false",
```

Comment:
Should we be applying [scripting best practices](https://docs.chocolatey.org/en-us/choco/commands/#scripting-integration-best-practices-style-guide) here?

Reply:
This is pretty close to directly taken from our docs. I'd hope that's okay, or ask what needs to be updated!